src/data_gradients/dataset_adapters/formatters/base.py

```diff
@@ -27,7 +27,7 @@ def get_n_image_channels(self, images: torch.Tensor) -> int:
         if self._n_image_channels is None:
             question = FixedOptionsQuestion(
                 question="Which dimension corresponds the image channel? ",
-                options={i: images.shape[i] for i in range(len(images.shape))},
+                options={dim: dim for dim in images.shape},
```

Comment:
Also, maybe you want to use collections.OrderedDict to guarantee the order of elements in that dictionary. Generally speaking order items in dict is not guaranteed to be the same across platforms (This was done intentionally  https://portingguide.readthedocs.io/en/latest/dicts.html https://stackoverflow.com/questions/20192950/why-items-order-in-a-dictionary-changed-in-python)

Reply:
Yeah, I get this, I think that's something to change with the `options` design and not with this specific usage.

### Some extra context
Basically `options` maps the description to the values. E.g. 
```python
question = FixedOptionsQuestion(
    question=f"{text_to_yellow('Which comes first')} in your annotations, the class id or the bounding box?",
    options={
         "Label comes first (e.g. [class_id, x1, y1, x2, y2])": True,
         "Bounding box comes first (e.g. [x1, y1, x2, y2, class_id])": False,
    },
)
```

The description is what the user "selects", and the value is ... well the value we use in the code.


**How does the user chose an answer ?** 
It would be crazy to expect the user input to be exactly `"Label comes first (e.g. [class_id, x1, y1, x2, y2])"`.

For this reason, we introduced options as index; 
Instead of 
```
Please choose
- Label comes first (e.g. [class_id, x1, y1, x2, y2])
- Bounding box comes first (e.g. [x1, y1, x2, y2, class_id])
Your selection >>> ...
```

we have something like
```
Please choose
1 | Label comes first (e.g. [class_id, x1, y1, x2, y2])
2 | Bounding box comes first (e.g. [x1, y1, x2, y2, class_id])
Your selection (Enter the corresponding number) >>> ...
```


TLDR; we have 3 pieces of information
- The description (that will be displayed)
- The actual value (less user friendly)
- The index (to simplify answering)
And currently, the index is computed on the fly when asking questions, and not saved anywhere (we save the value).

### Going back to this piece of code
You understood that option is (index -> value) which I agree is would be the intuitive logic, but what it represents is (description -> value), the index being, as explained, defined on the fly.

### How to change this ?
I guess we should have 
```
options: Dict[int, Option]
``` 
with 
```python
@dataclass
class Option
    value: ...
    descriptions: str
```

I think this would be much better. You ?

Reply:
Ok with `description -> value` this definitely have much more sense now :)
I'm good with this PR. 
Since it's anyway part of internal API i think we can live with existing `description -> value` notion as long as it is documented. Not pressure on refactoring this if it works